indico/modules/events/registration/models/registrations.py

```diff
@@ -697,3 +702,55 @@ def _set_checked_in_dt(target, value, *unused):
     @listens_for(Registration.transaction, 'set')
     def _set_transaction_id(target, value, *unused):
         value.registration = target
+
+
+class RegistrationTag(db.Model):
+    """Registration tag used to mark/filter registrations."""
+    __tablename__ = 'registration_tags'
+    __table_args__ = {'schema': 'event_registration'}
+
+    #: The ID of the object
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    #: The ID of the event where this tag was created
+    event_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.events.id'),
+        index=False,
+        nullable=False
+    )
+    #: Tag name
+    name = db.Column(
+        db.String,
+        nullable=False
+    )
+    #: Tag color (hex)
+    color = db.Column(
+        db.String,
+        nullable=False
+    )
+    #: The Event where this tag was created
+    event = db.relationship(
+        'Event',
+        lazy=True,
+        backref=db.backref(
+            'registration_tags',
+            lazy='dynamic'
+        )
+    )
+
+    @property
+    def locator(self):
+        return dict(self.event.locator, tag_id=self.id)
+
+
+class RegistrationRegistrationTag(db.Model):
```

Comment:
Search for `db.Table` to find some examples on how to define many to many mapping tables